### PR TITLE
chore: Unify python dependencies into requirements.in

### PR DESCRIPTION
**Pull request by rafiu007**
This is WIP PR which takes care of dependency unification for the api server.

Checklist ::
 - [x] Move any test specific packages from requirements.in to tests/requirements.in
 - [x] Move any direct pip installations from Dockerfile.* to requirements.in
 - [ ] Move any direct pip installations from tests/*.sh to tests/requirements.in
 - [x] Relative versioning in requirements.in & tests/requirements.in (no version pinning unless otherwise needed)
 - [x] Generate requirements.txt (not required) and tests/requirements.txt
 - [ ] (Not Applicable)Update setup.py to use requirement.in
 - [x] Cleanup unused dependencies from requirements.in and tests/requirements.in (using `pip install pipreqs`)

### CODE REVIEW

File: requirements.in
Comment (dgpatelgit) [inline requirements.in:2]: We need this flask and mod_wsgi and many other deps in actual docker images.

File: tests/requirements.in
Comment (dgpatelgit) [inline tests/requirements.in:10]: You can include development requirement file first and than add test dependencies.

File: requirements.in
Comment (dgpatelgit) [inline requirements.in:28]: Format is not valid, we need to have egg=<folder Path> and before that we need to tag this deps with version?
Same goes for other three deps below.

Example Syntax:
https://github.com/fabric8-analytics/fabric8-analytics-version-comparator.git@8a57ac7#egg=f8a_version_comparator

File: requirements.in
Comment (dgpatelgit) [inline requirements.in:13]: Do not have test dependencies in this file. Create a separate test requirement.in that takes all present in main requirement.in and add test deps on top. Refer to commit https://github.com/fabric8-analytics/fabric8-analytics-utils/commit/21ea911551248a4d485d1ac3e42c1ad7fa6849a0

**Comment by arajkumar** [comment]
@rafiu007 Could you please follow the template/checklist described in https://github.com/fabric8-analytics/fabric8-analytics-utils/pull/57#issue-454207573?

File: setup.py
Comment (deepak1725) [inline setup.py:10]: Not sure if we really need this..? requirement.in outputs requirements.txt 
It's `.txt` which ideally be used for further processing. 
Why `.txt` coz it has a list of dependencies and transitive in flat list format.

File: tests/requirements.txt
Comment (deepak1725) [inline tests/requirements.txt:49]: Please make sure while generating this file, your virtual env is activated and it has no dependencies installed previously. 
Then ideally the process should be
1. Create and Activate new Virtual Env
2. Add new Deps that are lying in Dockerfile or so in requirements.in
3. `pip-compile --output-file=requirements.txt requirements.in`
4. `pip install -r requirements.txt` to check any Errors.

**Review by deepak1725 — CHANGES_REQUESTED** [review]

File: tests/requirements.txt
Comment (rafiu007) [reply tests/requirements.txt:49]: I am checking directly on the cluster after building the image.Is that not okay?

File: setup.py
Comment (rafiu007) [reply setup.py:10]: I did this cause it was recommended. @dgpatelgit any idea why?

Comment (dgpatelgit) [reply setup.py:10]: @deepak1725 We need this change as suggested by the initial check list provided by Arun for this migration task.

**Review by dgpatelgit — COMMENTED** [review]
I have added check-list template to PR description, @rafiu007 can you check and update the list as per code changes?

File: requirements.in
Comment (arajkumar) [inline requirements.in:28]:
```suggestion
f8a_worker @ git+https://github.com/fabric8-analytics/fabric8-analytics-worker.git@<hash>#egg=f8a_worker
```

File: requirements.in
Comment (arajkumar) [inline requirements.in:31]: This is not required since it will be pulled by utils.

File: setup.py
Comment (arajkumar) [reply setup.py:10]: @deepak1725, requirement.txt will have pinned versions which would cause conflicts. However this is not required in this repo because it is not a installable library. It is an application.

File: tests/requirements.txt
Comment (arajkumar) [inline tests/requirements.txt:4]: Add `tests/requirements.in` and include main `requirements.in` in it.

Reference: https://github.com/fabric8-analytics/fabric8-analytics-utils/blob/5a5ce60c38c042a43733e7433aa01c71d16542f1/tests/requirements.in#L5

**Review by arajkumar — CHANGES_REQUESTED** [review]

File: requirements.in
Comment (rafiu007) [reply requirements.in:28]: @dgpatelgit After adding versions(commit hash) for the other three deps the docker build fails.

File: requirements.in
Comment (arajkumar) [inline requirements.in:30]:
```suggestion
f8a_worker-version @ git+https://github.com/fabric8-analytics/fabric8-analytics-worker.git@066c2f6#egg=f8a_worker
f8a_auth_version @ git+https://github.com/fabric8-analytics/fabric8-analytics-auth.git@5ff9438#egg=fabric8a_auth
f8a_utils @ git+https://github.com/fabric8-analytics/fabric8-analytics-utils.git@5a5ce60#egg=f8a_utils
```

File: requirements.in
Comment (arajkumar) [reply requirements.in]: It is failing because the mentioned eggs are incorrect. Please check the suggestion.

File: requirements.in
Comment (arajkumar) [inline requirements.in:24]:
```suggestion
f8a_worker @ git+https://github.com/fabric8-analytics/fabric8-analytics-worker.git@066c2f6#egg=f8a_worker
f8a_auth @ git+https://github.com/fabric8-analytics/fabric8-analytics-auth.git@5ff9438#egg=fabric8a_auth
```

File: tests/requirements.in
Comment (arajkumar) [inline tests/requirements.in:11]: Include only test related packages which are used currently.

**Review by arajkumar — CHANGES_REQUESTED** [review]

**Review by arajkumar — COMMENTED** [review]
Also Dockerfile.tests should be modified to install pip packages from tests/requirements.txt

File: tests/requirements.in
Comment (arajkumar) [reply tests/requirements.in:11]: Also Dockerfile.tests should be modified to install pip packages from tests/requirements.txt.

Comment (rafiu007) [reply tests/requirements.in:11]: Missed out tenacity somehow,removing it now.

Comment (arajkumar) [reply tests/requirements.in:11]: I see that quite few unused packages in both requirments.in and tests/requirements.in. You can use `pipreqs` to find only the used package in the source.
 `pip install pipreqs`

Comment (rafiu007) [reply tests/requirements.in:11]: @arajkumar I used pipreqs find packages.

Comment (rafiu007) [reply tests/requirements.in:11]: And I only included them in the requirements.in and then created requirements.txt with pip-compile.

**Comment by arajkumar** [comment]
Just started reviewing :)

**Comment by arajkumar** [comment]
Just started reviewing :)

**Review by arajkumar — APPROVED** [review]
lgtm

@rafiu007 Please wait for one more approval.

**Review by dgpatelgit — APPROVED** [review]
LGTM

**Review by deepak1725 — APPROVED** [review]
lgtm

**Comment by arajkumar** [comment]
Just started reviewing :)

**Pull request merged**
